components/engine/src/lib.rs

```diff
@@ -216,99 +216,70 @@ pub trait Peekable {
     }
 }
 
-#[derive(Clone, PartialEq)]
-enum SeekMode {
-    TotalOrder,
-    Prefix,
-}
-
-pub struct IterOption {
-    lower_bound: Option<Vec<u8>>,
-    upper_bound: Option<Vec<u8>>,
-    prefix_same_as_start: bool,
-    fill_cache: bool,
-    seek_mode: SeekMode,
-}
+pub struct IterOption(ReadOptions);
 
 impl IterOption {
     pub fn new(
         lower_bound: Option<Vec<u8>>,
         upper_bound: Option<Vec<u8>>,
         fill_cache: bool,
     ) -> IterOption {
-        IterOption {
-            lower_bound,
-            upper_bound,
-            prefix_same_as_start: false,
-            fill_cache,
-            seek_mode: SeekMode::TotalOrder,
+        let mut read_options = ReadOptions::new();
```

Comment:
So why not use ReadOptions directly?

Reply:
So we can't use like let opt = IterOptions::new(xxx).xxx().xxx().xxx().

Reply:
Got it. How about calling it `ReadOptionsBuilder`?